app/views/editor/level/modals/SaveBranchModal.coffee

```diff
@@ -0,0 +1,185 @@
+    
+    # insert all the Delta views for the systems/components which will form the branch
+    changeEls = @$el.find('.component-changes-stub')
+    for changeEl in changeEls
+      componentId = $(changeEl).data('component-id')
+      component = @componentsWithChanges.find((c) -> c.id is componentId)
+      @insertDeltaView(component, changeEl)
+
+    changeEls = @$el.find('.system-changes-stub')
+    for changeEl in changeEls
+      systemId = $(changeEl).data('system-id')
+      system = @systemsWithChanges.find((c) -> c.id is systemId)
+      @insertDeltaView(system, changeEl)
+      
+  insertDeltaView: (model, changeEl, headModel) ->
+    try
+      deltaView = new DeltaView({model: model, headModel, skipPaths: deltasLib.DOC_SKIP_PATHS})
+      @insertSubView(deltaView, $(changeEl))
+      return deltaView
+    catch e
+      console.error 'Couldn\'t create delta view:', e
+        
+  renderSelectedBranch: ->
+    # insert delta subviews for the selected branch, including the 'headComponent' which shows
+    # what, if any, conflicts the existing branch has with the client's local changes
+    
+    @removeSubView(view) for view in @selectedBranchDeltaViews if @selectedBranchDeltaViews
+    @selectedBranchDeltaViews = []
+    @renderSelectors('#selected-branch-col')
+    changeEls = @$el.find('#selected-branch-col .component-changes-stub')
+    for changeEl in changeEls
+      componentId = $(changeEl).data('component-id')
+      component = @selectedBranch.components.get(componentId)
+      targetComponent = @components.find((c) -> c.get('original') is component.get('original') and c.get('version').isLatestMajor)
+      preBranchSave = component.clone()
+      preBranchSave.markToRevert()
+      componentDiff = targetComponent.clone()
+      preBranchSave.set(componentDiff.attributes)
+      @selectedBranchDeltaViews.push(@insertDeltaView(preBranchSave, changeEl))
+
+    changeEls = @$el.find('#selected-branch-col .system-changes-stub')
+    for changeEl in changeEls
+      systemId = $(changeEl).data('system-id')
+      system = @selectedBranch.systems.get(systemId)
+      targetSystem = @systems.find((c) -> c.get('original') is system.get('original') and c.get('version').isLatestMajor)
+      preBranchSave = system.clone()
+      preBranchSave.markToRevert()
+      systemDiff = targetSystem.clone()
+      preBranchSave.set(systemDiff.attributes)
+      @selectedBranchDeltaViews.push(@insertDeltaView(preBranchSave, changeEl))
+
+  onClickBranch: (e) ->
+    $(e.currentTarget).closest('.list-group').find('.active').removeClass('active')
+    $(e.currentTarget).addClass('active')
+    branchCid = $(e.currentTarget).data('branch-cid')
+    @selectedBranch = if branchCid then @branches.get(branchCid) else null
+    @renderSelectedBranch()
+
+  onClickStashBranchButton: (e) ->
+    @deleteSavedChanges(e, {deleteSavedChanges: true})
+
+  onClickSaveBranchButton: (e) ->
+    @deleteSavedChanges(e, {deleteSavedChanges: false})
```

Comment:
This call is really confusing/seems paradoxical. Sounds like something here should be renamed (probably the method)